superset/db_engine_specs/base.py

```diff
@@ -967,25 +980,27 @@ def make_label_compatible(cls, label: str) -> Union[str, quoted_name]:
         return label_mutated
 
     @classmethod
-    def get_sqla_column_type(cls, type_: Optional[str]) -> Optional[TypeEngine]:
+    def get_sqla_column_type(
+        cls, column_type: Optional[str]
+    ) -> Tuple[Union[TypeEngine, utils.GenericDataType, None]]:
         """
         Return a sqlalchemy native column type that corresponds to the column type
         defined in the data source (return None to use default type inferred by
         SQLAlchemy). Override `column_type_mappings` for specific needs
         (see MSSQL for example of NCHAR/NVARCHAR handling).
 
-        :param type_: Column type returned by inspector
+        :param column_type: Column type returned by inspector
         :return: SqlAlchemy column type
         """
-        if not type_:
-            return None
-        for regex, sqla_type in cls.column_type_mappings:
-            match = regex.match(type_)
+        if not column_type:
+            return None, None
+        for regex, sqla_type, generic_type in cls.column_type_mappings:
+            match = regex.match(column_type)
             if match:
                 if callable(sqla_type):
-                    return sqla_type(match)
-                return sqla_type
-        return None
+                    return sqla_type(match), generic_type
+                return sqla_type, generic_type
+        return None, None
```

Comment:
For cases where we want to return an empty value (=no match was made), I'd perhaps prefer returning a pure `None` instead of a `Tuple` with `None`s.